source/resolve-conflicts.ts

```diff
@@ -0,0 +1,109 @@
+/// <reference types="codemirror" />
+
+interface CodeMirrorInstance extends CodeMirror.Editor, CodeMirror.Doc {}
+
+const select: typeof document.querySelector = document.querySelector.bind(document);
+const editor: CodeMirrorInstance = select<any>('.CodeMirror').CodeMirror;
+
+// Event fired when each file is loaded
+editor.on('swapDoc', () => setTimeout(addWidget, 1));
+
+// Restore widget on undo
+editor.on('changes', (_, [firstChange]) => {
+	if (firstChange.origin === 'undo' && firstChange.text[0].startsWith('<<<<<<<')) {
+		addWidget();
+	}
+});
+
+function getLineNumber(lineChild: Element) {
+	return Number(
+		lineChild
+			.closest('.CodeMirror-gutter-wrapper, .CodeMirror-linewidget')
+			.parentElement
+			.querySelector('.CodeMirror-linenumber')
+			.textContent
+	) - 1;
+}
+
+function appendLineInfo(lineHandle: CodeMirror.LineHandle, text: string) {
+	// Only append text if it's not already there
+	if (!lineHandle.text.includes(text)) {
+		const line = (lineHandle as any).lineNo();
```

Comment:
The `CodeMirror.LineHandle` is incomplete, @sindresorhus how can I extend it and avoid `as any`?

Reply:
What you've done in https://github.com/sindresorhus/refined-github/pull/1882/commits/5c377fa23b8c20ed6f7281c1fe467c6366fd40f1 is correct. Ideally, you would submit a fix to DefinitelyTyped too.